File: Leetcode/Daily/07-2025/Python3/15.py
# Problem: 3136. Valid Word (Easy)
# Link: https://leetcode.com/problems/valid-word

class Solution:
    def isValid(self, word: str) -> bool:
        if len(word) < 3:
            return False

        vowel = False
        consonant = False

        for ch in word:
            if '0' <= ch and ch <= '9': 
                continue
            elif 'A' <= ch and ch <= 'Z':
                if ch == 'A' or ch == 'E' or ch == 'I' or ch == 'O' or ch == 'U':
                    vowel = True
                else: 
                    consonant = True
            elif 'a' <= ch and ch <= 'z':
                if ch == 'a' or ch == 'e' or ch == 'i' or ch == 'o' or ch == 'u': 
                    vowel = True
                else: 
                    consonant = True
            else: 
                return False

        return vowel and consonant

File: Leetcode/Daily/07-2025/Java/15.java
// Problem: 3136. Valid Word (Easy)
// Link: https://leetcode.com/problems/valid-word

class Solution {
    public boolean isValid(String word) {
        if (word.length() < 3) return false;

        boolean vowel = false, consonant = false;
        int n = word.length();

        for (int i = 0; i < n; i++) {
            char ch = word.charAt(i);
            if ('0' <= ch && ch <= '9') continue;
            else if ('A' <= ch && ch <= 'Z') {
                if (ch == 'A' || ch == 'E' || ch == 'I' || ch == 'O' || ch == 'U') vowel = true;
                else consonant = true;
            } else if ('a' <= ch && ch <= 'z') {
                if (ch == 'a' || ch == 'e' || ch == 'i' || ch == 'o' || ch == 'u') vowel = true;
                else consonant = true;
            } else return false;
        }

        return vowel && consonant;        
    }
}